sys/net/npf/npf_ctl.c

```diff
@@ -378,6 +378,31 @@ npf_mk_singlerule(npf_t *npf, const nvlist_t *req, nvlist_t *resp,
 		npf_rule_setcode(rl, type, bc, clen);
 	}
 
+	/* user and group ids filt option if set */
+	if (nvlist_exists_number_array(req, "r_user")) {
+		size_t uitems;
+		struct r_uid r_uid;
+		const uint64_t *uid = nvlist_get_number_array(req, "r_user", &uitems);
+		KASSERT(uitems == 3);
+
+		r_uid.uid[0] = (uint32_t)uid[0];
```

Comment:
these should be (uid_t) not (uint32_t).

Reply:
but... you are using so much uint32_t and uid_t/gid_t interchangeably, so why don't you just call it id in the structure, make it uint32_t and then merge all the duplicated code. You can put in a comment why (both uid_t and gid_t and uint32_t and using uint32_t saves a lot of code duplication).